controller/src/main/java/org/jboss/as/controller/registry/ManagementResourceRegistration.java

```diff
@@ -310,7 +311,31 @@
      */
     class Factory {
 
+        public static final Factory DEFAULT = new Factory();
+
+        private final ProcessType processType;
+
         private Factory() {
+            processType = null;
+        }
+
+        private Factory(ProcessType processType) {
+            this.processType = processType;
+        }
+
+        /**
+         * Returns a ManagementResourceRegistration's Factory that will use the specified {@code processType}
+         * to determine whether resource metrics are registered or not.
+         *
+         * If the {@code processType} id {@code null}, metrics are <em>always</em> registered.
+         *
+         * @param processType can be {@code null}
+
+         * @return a Factory which creates ManagementResourceRegistration that
+         * dynamically determine whether resource metrics are actually registered
+         */
+        public static Factory forServer(ProcessType processType) {
```

Comment:
This is a bit of a nitpick, but personally, I don't like the name 'forServer'. When I saw the first usage, I got curious if there was a forDomain() method as well.